Tasks/Common/ServiceFabricHelpers/Connect-ServiceFabricClusterFromServiceEndpoint.ps1

```diff
@@ -43,15 +43,22 @@ function Get-AadSecurityToken
 
     try
     {
-        # Acquiring a token using UserCredential implies a non-interactive flow. No credential prompts will occur.
-        $accessToken = $authContext.AcquireToken($clusterApplicationId, $clientApplicationId, $userCredential).AccessToken
+        if($script:RefreshToken)
+        {
+            $token = $authContext.AcquireTokenByRefreshToken($RefreshToken, $clientApplicationId)
+        }
+        else
+        {
+            # Acquiring a token using UserCredential implies a non-interactive flow. No credential prompts will occur.
```

Comment:
Creation of object clientApplicationId and userCredential can be moved to this ELSE to avoid unnecessary creation of these objects in case if code goes to IF part. 
But I'm intentionally not moving it here since this is more readable and its very very rare that code flow will go to IF statement.

Reply:
Hmm. Then why not get the token using userCredential in the IF case also. Which basically means don;t use refresh token at all - everytime just use userCredential to get token. Cleanest code!
But we should check performance of getting access token when not using refresh token.

Reply:
It is 'MUST' recommended to use RefreshToken always to acquire additional access tokens after the current access token expires.